### ISSUE TITLE
Can't build or run tests

### ISSUE THREAD

**Issue by asvvvad** [therecipe/qt#1105]
I want to deploy to android so I downloaded the big official version and installed it successfully and set the environment variables accordingly

Later I followed the [wiki tutorial](https://github.com/therecipe/qt/wiki/Installation-on-Linux) and started cloning the repo using `go get -v -tags=no_env github.com/therecipe/qt/cmd/...` as specified
However after a long wait for it to download I found it's downloading the build environments for every platform which is like a gigabyte or more

I'm on Linux ubuntu 18 based distro ElementaryOS using go version go1.14.2 linux/amd64 and Qt5.14.2

Edit: Yes `GO111MODULE` is set to on since that specific tutorial didn't mention it, should it be on? but if I did will it download just the linux env? is that needed since I have the official version of qt?

**Comment by asvvvad** [comment]
So i tried it with GO111MODULE off and it went okay and managed to install it
But once i try to `qtsetup full android` I get this:
```
INFO[0000] running: 'qtsetup prep'                      
WARN[0000] failed to create qtrcc symlink in your PATH (/sbin/qtrcc); please use /home/asvvvad/go/bin/qtrcc instead 
WARN[0000] failed to create qtmoc symlink in your PATH (/sbin/qtmoc); please use /home/asvvvad/go/bin/qtmoc instead 
WARN[0000] failed to create qtminimal symlink in your PATH (/sbin/qtminimal); please use /home/asvvvad/go/bin/qtminimal instead 
WARN[0000] failed to create qtdeploy symlink in your PATH (/sbin/qtdeploy); please use /home/asvvvad/go/bin/qtdeploy instead 
INFO[0000] running: 'qtsetup check android' [docker=false] [vagrant=false] 
INFO[0000] GOOS:                        'linux'         
INFO[0000] GOARCH:                      'amd64'         
INFO[0000] GOVERSION:                   'go1.14.2'      
INFO[0000] GOROOT:                   *  '/usr/local/go' 
INFO[0000] GOPATH:                   *  '/home/asvvvad/go' 
INFO[0000] GOBIN:                       '/home/asvvvad/go/bin' 
INFO[0000] GOMOD:                       '/dev/null'     
INFO[0000] QT_HASH:                     ''              
INFO[0000] QT_API:                   *  ''              
INFO[0000] QT_VERSION:               *  '5.14.2'        
INFO[0000] QT_DIR:                   *  '/home/asvvvad/Qt5.14.2' 
INFO[0000] QT_STUB:                     'false'         
INFO[0000] QT_DEBUG:                    'false'         
INFO[0000] QT_QMAKE_DIR:                ''              
INFO[0000] QT_WEBKIT:                   'false'         
INFO[0000] QT_STATIC:                   'false'         
INFO[0000] QT_GEN_TSD:                  'false'         
INFO[0000] QT_GEN_OPENGL:               'false'         
INFO[0000] QT_GEN_QUICK_EXTRAS:         'false'         
INFO[0000] QT_RESOURCES_BIG:            'false'         
INFO[0000] QT_NOT_CACHED:               'false'         
INFO[0000] JDK_DIR:                     '/usr/lib/jvm/java-8-openjdk-amd64' 
INFO[0000] ANDROID_SDK_DIR:          *  '.'             
INFO[0000] ANDROID_NDK_DIR:          *  '/home/asvvvad/.local/android-ndk-r18b' 
INFO[0000] running: 'qtsetup generate android' [docker=false] [vagrant=false] 
panic: runtime error: index out of range [0] with length 0

goroutine 23 [running]:
github.com/therecipe/qt/internal/utils.QT_API_NUM(0xc00001a08b, 0x6, 0xc0000f1940)
	/home/asvvvad/go/src/github.com/therecipe/qt/internal/utils/env.go:90 +0x13c
github.com/therecipe/qt/internal/binding/parser.LoadModule(0x7c400f, 0xd, 0x0)
	/home/asvvvad/go/src/github.com/therecipe/qt/internal/binding/parser/parser.go:101 +0x1e5
github.com/therecipe/qt/internal/binding/parser.LoadModules.func1(0xc0000abe20, 0xc0000bf600, 0x29, 0x29, 0xc0000abe30, 0x1, 0x7c400f, 0xd)
	/home/asvvvad/go/src/github.com/therecipe/qt/internal/binding/parser/parser.go:39 +0x39
created by github.com/therecipe/qt/internal/binding/parser.LoadModules
	/home/asvvvad/go/src/github.com/therecipe/qt/internal/binding/parser/parser.go:38 +0x170
```
 I can't proceed from here

**Comment by asvvvad** [comment]
Same for linux:
```
INFO[0000] running: 'qtsetup generate linux' [docker=false] [vagrant=false] 
panic: runtime error: index out of range [0] with length 0

goroutine 59 [running]:
github.com/therecipe/qt/internal/utils.QT_API_NUM(0xc00001808b, 0x6, 0xc000048940)
	/home/asvvvad/go/src/github.com/therecipe/qt/internal/utils/env.go:90 +0x13c
github.com/therecipe/qt/internal/binding/parser.LoadModule(0x7c5adf, 0xd, 0x0)
	/home/asvvvad/go/src/github.com/therecipe/qt/internal/binding/parser/parser.go:101 +0x1e5
github.com/therecipe/qt/internal/binding/parser.LoadModules.func1(0xc0000a8720, 0xc0000bedc0, 0x29, 0x29, 0xc0000a8730, 0x28, 0x7c5adf, 0xd)
	/home/asvvvad/go/src/github.com/therecipe/qt/internal/binding/parser/parser.go:39 +0x39
created by github.com/therecipe/qt/internal/binding/parser.LoadModules
	/home/asvvvad/go/src/github.com/therecipe/qt/internal/binding/parser/parser.go:38 +0x170
```

**Comment by asvvvad** [comment]
I looked in the source at line 90 and found it have to do with the QT_API it was set 5.14.0 then removed it entirely restarted the terminal after a lot of failed tries and it passed that point but only to stop by another error at `INFO[0019] generating cgo qt/core`
```
ERRO[0019] failed to run command                         _func=RunCmd cmd="go list -mod=vendor -find -e -f {{.Dir}} github.com/therecipe/qt" dir=/dev env="LS_COLORS=rs=0:di=01;34:ln=01;36:mh=00:pi=40;33:so=01;35:do=01;35:bd=40;33;01:cd=40;33;01:or=40;31;01:mi=00:su=37;41:sg=30;43:ca=30;41:tw=30;42:ow=34;42:st=37;44:ex=01;32:*.tar=01;31:*.tgz=01;31:*.arc=01;31:*.arj=01;31:*.taz=01;31:*.lha=01;31:*.lz4=01;31:*.lzh=01;31:*.lzma=01;31:*.tlz=01;31:*.txz=01;31:*.tzo=01;31:*.t7z=01;31:*.zip=01;31:*.z=01;31:*.Z=01;31:*.dz=01;31:*.gz=01;31:*.lrz=01;31:*.lz=01;31:*.lzo=01;31:*.xz=01;31:*.zst=01;31:*.tzst=01;31:*.bz2=01;31:*.bz=01;31:*.tbz=01;31:*.tbz2=01;31:*.tz=01;31:*.deb=01;31:*.rpm=01;31:*.jar=01;31:*.war=01;31:*.ear=01;31:*.sar=01;31:*.rar=01;31:*.alz=01;31:*.ace=01;31:*.zoo=01;31:*.cpio=01;31:*.7z=01;31:*.rz=01;31:*.cab=01;31:*.wim=01;31:*.swm=01;31:*.dwm=01;31:*.esd=01;31:*.jpg=01;35:*.jpeg=01;35:*.mjpg=01;35:*.mjpeg=01;35:*.gif=01;35:*.bmp=01;35:*.pbm=01;35:*.pgm=01;35:*.ppm=01;35:*.tga=01;35:*.xbm=01;35:*.xpm=01;35:*.tif=01;35:*.tiff=01;35:*.png=01;35:*.svg=01;35:*.svgz=01;35:*.mng=01;35:*.pcx=01;35:*.mov=01;35:*.mpg=01;35:*.mpeg=01;35:*.m2v=01;35:*.mkv=01;35:*.webm=01;35:*.ogm=01;35:*.mp4=01;35:*.m4v=01;35:*.mp4v=01;35:*.vob=01;35:*.qt=01;35:*.nuv=01;35:*.wmv=01;35:*.asf=01;35:*.rm=01;35:*.rmvb=01;35:*.flc=01;35:*.avi=01;35:*.fli=01;35:*.flv=01;35:*.gl=01;35:*.dl=01;35:*.xcf=01;35:*.xwd=01;35:*.yuv=01;35:*.cgm=01;35:*.emf=01;35:*.ogv=01;35:*.ogx=01;35:*.aac=00;36:*.au=00;36:*.flac=00;36:*.m4a=00;36:*.mid=00;36:*.midi=00;36:*.mka=00;36:*.mp3=00;36:*.mpc=00;36:*.ogg=00;36:*.ra=00;36:*.wav=00;36:*.oga=00;36:*.opus=00;36:*.spx=00;36:*.xspf=00;36: LESSCLOSE=/usr/bin/lesspipe %s %s ANDROID_HOME=/home/asvvvad/.local/sdk XDG_MENU_PREFIX=gnome- LANG=en_US.UTF-8 GDM_LANG=en_US PANTHEON_TERMINAL_ID=0 DISPLAY=:0 QT_STYLE_OVERRIDE=adwaita GTK_CSD=1 COLORTERM=truecolor QT_VERSION=5.14.2 JAVA_HOME=/usr/lib/jvm/java-8-openjdk-amd64 XDG_VTNR=7 GIO_LAUNCHED_DESKTOP_FILE_PID=14389 GTK3_MODULES=pantheon-filechooser-module SSH_AUTH_SOCK=/run/user/1001/keyring/ssh MANDATORY_PATH=/usr/share/gconf/pantheon.mandatory.path XDG_SESSION_ID=c2 XDG_GREETER_DATA_DIR=/var/lib/lightdm-data/asvvvad USER=asvvvad GO111MODULE=on DESKTOP_SESSION=pantheon CPLY_TOKEN=GtWo7LIZkd22NTn9CbhXAVXA4zISeqCkpOXwVSsj-WVpUt1sDC6EulqXOr-00Gwc GOPATH=/home/asvvvad/go GRADLE_HOME=/opt/gradle/gradle-5.6.2 IGTY_CONFIG=/home/asvvvad/.igty DEFAULTS_PATH=/usr/share/gconf/pantheon.default.path QT_QPA_PLATFORMTHEME=gtk3 PWD=/home/asvvvad HOME=/home/asvvvad GOROOT=/usr/local/go SSH_AGENT_PID=1509 QT_ACCESSIBILITY=1 ANDROID_NDK_ROOT=/home/asvvvad/.local/android-ndk-r18b LIBVIRT_DEFAULT_URI=qemu:///system XDG_SESSION_TYPE=x11 XDG_DATA_DIRS=/usr/share/gnome:/usr/share/pantheon:/home/asvvvad/.local/share/flatpak/exports/share:/var/lib/flatpak/exports/share:/usr/local/share:/usr/share:/var/lib/snapd/desktop QT_DIR=/home/asvvvad/Qt5.14.2 ANDROID_SDK_DIR= GSETTINGS_SCHEMA_DIR=/home/asvvvad/data XDG_SESSION_DESKTOP=pantheon GTK_MODULES=gail:atk-bridge VTE_VERSION=5202 TERM=xterm-256color SHELL=/bin/bash XDG_SEAT_PATH=/org/freedesktop/DisplayManager/Seat0 QT_IM_MODULE=ibus XMODIFIERS=@im=ibus XDG_CURRENT_DESKTOP=Pantheon GPG_AGENT_INFO=/run/user/1001/gnupg/S.gpg-agent:0:1 GIO_LAUNCHED_DESKTOP_FILE=/usr/share/applications/io.elementary.terminal.desktop XDG_SEAT=seat0 SHLVL=1 LANGUAGE=en_US PROMPT_COMMAND=__bp_precmd_invoke_cmd; dbus-send --type=method_call --session --dest=io.elementary.terminal /io/elementary/terminal io.elementary.terminal.ProcessFinished string:$PANTHEON_TERMINAL_ID string:\"$(history 1 | cut -c 8-)\" int32:$__bp_last_ret_value >/dev/null 2>&1; __bp_interactive_mode; GDMSESSION=pantheon GNOME_DESKTOP_SESSION_ID=this-is-deprecated ANDROID_NDK_DIR=/home/asvvvad/.local/android-ndk-r18b LOGNAME=asvvvad DBUS_SESSION_BUS_ADDRESS=unix:path=/run/user/1001/bus XDG_RUNTIME_DIR=/run/user/1001 XAUTHORITY=/home/asvvvad/.Xauthority XDG_SESSION_PATH=/org/freedesktop/DisplayManager/Session0 XDG_CONFIG_DIRS=/etc/xdg/xdg-pantheon:/etc/xdg PATH=/home/asvvvad/.config/composer/vendor/bin:/home/asvvvad/.local/bin:/home/asvvvad/.local/flutter/bin:/home/asvvvad/go/bin:/home/asvvvad/.local/sdk/tools:/home/asvvvad/.local/sdk/platform_tools/bin::/home/asvvvad/.local/sdk/platform-tools:/opt/gradle/gradle-5.6.2/bin:/usr/lib/ccache:/home/asvvvad/.local/bin:/home/asvvvad/.config/composer/vendor/bin:/home/asvvvad/.local/bin:/home/asvvvad/.local/flutter/bin:/home/asvvvad/go/bin:/home/asvvvad/.local/sdk/tools:/home/asvvvad/.local/sdk/platform_tools/bin::/home/asvvvad/.local/sdk/platform-tools:/opt/gradle/gradle-5.6.2/bin:/home/asvvvad/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/home/asvvvad/Qt5.14.2/5.14.2/gcc_64/bin/ SESSION_MANAGER=local/elementary:@/tmp/.ICE-unix/1405,unix/elementary:/tmp/.ICE-unix/1405 LESSOPEN=| /usr/bin/lesspipe %s _=/home/asvvvad/go/bin/qtsetup CGO_ENABLED=0" error="exit status 1" name=utils.GoQtPkgPath
go: cannot find main module; see 'go help modules'
```

**Comment by asvvvad** [comment]
I passed that by logging by writing a bash script that export all the necessary variables and run the command and ran that with sudo but it does the same thing when it start running the tests or when i try to qtdeply build something:`
```
runtime/cgo
# runtime/cgo
In file included from _cgo_export.c:3:
/usr/include/stdlib.h:25:10: fatal error: 'bits/libc-header-start.h' file not found
```

**Comment by asvvvad** [comment]
fixed by using 5.13.0
